src/main/java/net/smartcosmos/dto/metadata/MetadataResponse.java

```diff
@@ -19,26 +21,25 @@
     @Setter(AccessLevel.NONE)
     private int version = VERSION; // just in case there is a default constructor sometime
 
-    private final String dataType;
-    private final String entityReferenceType;
-    private final String key;
-    private final Long lastModifiedTimestamp;
-    private final String rawValue;
-    private final String referenceUrn;
-    private final String urn;
-    private final String moniker;
+    private final String id;
+    private final String ownerType;
+    private final String ownerId;
+    private Map<String, Object> metadata;
+    private final String tenantId;
 
     @Builder
-    @ConstructorProperties({"dataType", "entityReferenceType", "key", "lastModifiedTimestamp", "rawValue", "referenceUrn", "urn", "moniker"})
-    public MetadataResponse(String dataType, String entityReferenceType, String key, Long lastModifiedTimestamp, String rawValue, String referenceUrn, String urn, String moniker) {
-        this.dataType = dataType;
-        this.entityReferenceType = entityReferenceType;
-        this.key = key;
-        this.lastModifiedTimestamp = lastModifiedTimestamp;
-        this.rawValue = rawValue;
-        this.referenceUrn = referenceUrn;
-        this.urn = urn;
-        this.moniker = moniker;
+    @ConstructorProperties({"id", "ownerType", "ownerId", "metadata", "tenantId"})
```

Comment:
If we do that, we can't use default values, because the Lombok builders ignores what is set on the field level. :(

That's why I had to delombok the `@AllArgsConstructor`.